File: Anul 2/Metode Avansate de Programare/Seminarii/sem11-taskrunner/taskrunner/taskrunner/runner/PrinterTaskRunner.cs
using taskrunner.utils;

namespace taskrunner.runner;

public class PrinterTaskRunner : AbstractTaskRunner
{
    public PrinterTaskRunner(ITaskRunner runner)
        : base(runner)
    {
    }

    public override void ExecuteOneTask()
    {
        base.ExecuteOneTask();
        DecorateExecuteOneTask();
    }

    public void DecorateExecuteOneTask()
    {
        Console.WriteLine(
            "task executat la: " +
            DateTime.Now.ToString(Constants.DateTimeFormatter)
        );
    }
}

File: Anul 2/Metode Avansate de Programare/Seminarii/Seminar 1/src/runner/PrinterTaskRunner.java
package runner;

public class PrinterTaskRunner extends AbstractTaskRunner
{
    public PrinterTaskRunner(TaskRunner runner)
    {
        super(runner);
    }

    @Override
    public void executeOneTask()
    {
        System.out.println("Executing one task");
        super.executeOneTask();
        System.out.println("Task executed at " + java.time.LocalDateTime.now());
    }
}
